source/extensions/quic_listeners/quiche/envoy_quic_proof_source.h

```diff
@@ -0,0 +1,35 @@
+#include "extensions/quic_listeners/quiche/envoy_quic_fake_proof_source.h"
+#include "extensions/quic_listeners/quiche/quic_transport_socket_factory.h"
+
+namespace Envoy {
+namespace Quic {
+
+class EnvoyQuicProofSource : public EnvoyQuicFakeProofSource,
```

Comment:
Having a real proof source inherit from a fake one seems strange, can we change it to inherit from quic::ProofSource directly?

Reply:
GetProof() is still fake, and once I address all the TODOs in the fake one, I will rename it to be a base class for EnvoyQuicProofSource and TestProofSource.

Reply:
In that case, please address the TODOs and renames as soon as this PR lands.

Reply:
+1!